ckanext/fairdatapoint/profiles.py

```diff
@@ -18,12 +18,16 @@ def _convert_extras_to_declared_schema_fields(dataset_dict):
     data_dict = {'type': dataset_type}
     full_schema_dict = toolkit.get_action('scheming_dataset_schema_show')(context, data_dict)
 
-    dataset_field_list = [x.get('field_name') for x in full_schema_dict.get('dataset_fields', [])]
+    dataset_field_list = { x.get('field_name') : x.get('preset') for x in full_schema_dict.get('dataset_fields', []) }
 
     # Populate the declared schema fields, if they are present in the extras
     for extra_dict in dataset_dict.get('extras', []):
         if extra_dict.get('key') in dataset_field_list:
```

Comment:
this `extra_dict.get('key')` is calculated multiple times, please consider a varible and maybe a check if it is not none

Reply:
Fixed.